docs/content/documentation/systems/Kernels/phase_field/KKSMultiACBulkF.md

```diff
@@ -1,8 +1,58 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
 # KKSMultiACBulkF
 !description /Kernels/KKSMultiACBulkF
 
+KKS Allen-Cahn kernel for the terms without a direct composition dependence.
+
+### Residual
+
+$$
+R=-\frac{dh}{d\eta}(F_a-F_b)+w\frac{dg}{d\eta}.
+$$
+
+### Jacobian
+
+#### On-diagonal
+
+We are looking for the $\frac \partial{\partial u_j}$ derivative of $R$, where
+$u\equiv\eta$. We need to apply the chain rule and will again only keep terms
+with the $\frac{\partial \eta}{\partial u_j}\frac{\partial}{\partial \eta}=\phi_j \frac{\partial}{\partial\eta}$
+derivative.
+
+$$
+\begin{eqnarray*}
+J &=& -\phi_j \frac{\partial}{\partial \eta}\left( \frac{dh}{d\eta}(F_a-F_b) \right) + w \phi_j \frac{\partial}{\partial \eta}\frac{dg}{d\eta} \\
+&=&-\frac{d^2h}{d\eta^2}\phi_j(F_a-F_b) + w\frac{d^2g}{d\eta^2}\phi_j \\
+\end{eqnarray*}
+$$
+
+(The implicit dependence of $F_a(c_a)$ and $F_b(c_a)$ on $\eta$ through $c_a(c,\eta)$
+and $c_b(c,\eta)$ does not contribute to the Jacobian, so
+$\frac{\partial F_a}{\partial \eta} = \frac{\partial F_a}{\partial \eta} = 0)$.
+
+#### Off-Diagonal
+
+The off-diagonal components are calculated for any other variables that $F_a$
+and $F_b$ depend on. For example, for $c_a$:
+
+$$
+J = \frac{dh}{d\eta}\left( \frac{\partial F_a}{\partial c_a} - \frac{\partial F_b}{\partial c_a}\right)\phi_j
+$$
+
+$\frac{\partial F_b}{\partial c_a} = 0$ in the KKS formulation, so this term
+would not need to be included if $c_a$ was the only variable $F_a$ depended on.
+However, the code calculates derivatives with respect to all variables that
+$F_a$ and $F_b$ depend on in a general way so that the Jacobian entries for other
+dependencies are correctly computed using the same piece of code. For example,
+both $F_a$ and $F_b$ could depend on temperature $T$, in which case
+
+$$
+J = \frac{dh}{d\eta}\left( \frac{\partial F_a}{\partial T} - \frac{\partial F_b}{\partial T}\right)\phi_j
```

Comment:
Assuming the non-linear variable is $\eta_1$, for non-linear variables other than $\eta_i$, for example $c_1$, the off-diagonal Jacobian is
$$
J = \phi_j \frac{d h_1}{d \eta_1} \frac{d F_1}{d c_1}